iquip/apps/scheduler.py

```diff
@@ -0,0 +1,327 @@
+        elif idx < row:
+            self.experimentData = (self.experimentData[:idx] + self.experimentData[idx+1:row] +
+                           [self.experimentData[idx]] + self.experimentData[row:])
+
+        # TODO(giwon2004): emit signal for change of priority through artiq-proxy.
+        return True
+
+    def flags(self, index: QModelIndex) -> int:
+        """Overrided."""
+        defaultFlags = Qt.ItemIsSelectable | Qt.ItemIsEnabled
+        if index.isValid():
+            return Qt.ItemIsEditable | Qt.ItemIsDragEnabled | defaultFlags
+        return Qt.ItemIsDropEnabled | defaultFlags
+
+
+class ExperimentDelegate(QAbstractItemDelegate):
+    """Delegate for displaying the layout of each data in the experiment list.
+
+    TODO(giwon2004): Enabling buttons when displayed using QAbstractDelegate.
+    """
+
+    def paint(self,
+        painter: QPainter,
+        option: QStyleOptionViewItem,
+        index: QModelIndex):
+        """Overridden."""
+        info = index.data(Qt.DisplayRole)
+        expView = ExperimentView(info, self.parent())
+        expView.resize(option.rect.size())
+        painter.save()
+        painter.translate(option.rect.topLeft())
+        expView.render(painter)
+        painter.restore()
+
+    def sizeHint(self, option: QStyleOptionViewItem, index: QModelIndex) -> QSize:
+        """Overridden."""
+        info = index.data(Qt.DisplayRole)
+        expView = ExperimentView(info, self.parent())
+        return expView.sizeHint()
+
+
+class SchedulerApp(qiwis.BaseApp):
+    """App for displaying the submitted experiment queue.
+
+    Attributes:
+        schedulerFrame: The frame that shows the submitted experiment queue.
+    """
+
+    def __init__(self, name: str, parent: Optional[QObject] = None):
+        """Extended."""  
+        super().__init__(name, parent=parent)
+        self.schedulerFrame = SchedulerFrame()
+        # TODO(giwon2004): Below are for testing before connecting to artiq-proxy.
+        self.addExperiment(ExperimentInfo("exp1", {"rid": 9, "priority": 3}))
+        self.addExperiment(ExperimentInfo("exp2", {"rid": 10, "priority": 3}))
+        self.addExperiment(ExperimentInfo("exp3", {"rid": 11, "priority": 4}))
+        self.runExperiment(ExperimentInfo("exp4", {"rid": 12, "priority": 2}))
+
+    # TODO(giwon2004): Below are called by the signal from artiq-proxy.
+    def runExperiment(self, info: Optional[ExperimentInfo] = None):
+        """Sets the experiment onto 'currently running' section.
+
+        Args:
+            info: The experiment information.
+        """
+        self.schedulerFrame.expRun.updateInfo(info)
+
+    def addExperiment(self, info: ExperimentInfo):
+        """Adds the experiment to 'queued experiments' section (expList).
+
+        Args:
+            info: The experiment information.
+        """
+        self.schedulerFrame.model.experimentData.append(info)
```

Comment:
Isn't the type of `experimentData` `List[ExperimentView]`?
Thus, should be the parameter of `append()` an `ExperimentView` instance, not `ExperimentInfo`?

Reply:
I don't know about `QAbstractListModel` well..
Due to `ExperimentDelegate`, does it work well?